flexmeasures/ui/templates/crud/asset.html

```diff
@@ -853,16 +1042,260 @@ <h5 class="card-title">${sensor.name}</h5>
+                </div>
+                ` : ""}
+
+                ${isString ? `
+                <div class="shadow-sm border border-secondary p-2">
+                    <span class="fw-light fs-5">${newFieldValue ? newFieldValue : "<i>Not Set</i>"}</span>
+                </div>
+                ` : ""}
+            </div>
+        `;
+    }
+
+    async function renderFlexContextForm() {
+        flexContextFormEle.innerHTML = "";
+        
+        for (const [key, value] of Object.entries(assetFlexContext)) {
+            if (value !== null){
+                flexContextFormEle.innerHTML += await renderFlexContextInputField(key);
+            }
+        }
+
+        renderFlexSelectOptions();
+    }
+
+    async function udpateFlexContextFieldValue(dataType, sensorId) {
+        const powerCapacityInputValue = document.getElementById("power-capacity").value;
+        let highlightedCard = document.querySelector('.border-on-click');
+        if (!highlightedCard) {
+            showToast("Please select a field to update", "error");
+            return;
+        }
+        // get card id and remove '-control' at the end
+        const cardId = highlightedCard.id;
+        const flexId = cardId.slice(0, -8);
+        if (dataType === "power") {
+            assetFlexContext[flexId] = powerCapacityInputValue;
+        } else if (dataType === "sensor"){
+            if (flexId === "inflexible-device-sensors") {
+                assetFlexContext[flexId].push(sensorId);
+            } else {
+                assetFlexContext[flexId] = {sensor: sensorId};
+            }
+        }
+        highlightedCard = await renderFlexContextInputField(flexId);
+        flexContextFormEle.innerHTML = flexContextFormEle.innerHTML.replace(document.getElementById(`${flexId}-control`).outerHTML, highlightedCard);
+        await selectFlexField(document.getElementById(`${flexId}-control`) );
+    }
+
+    async function addFlexContextField(fieldName) {
+        assetFlexContext[fieldName] = "";
+        flexInputsContainer.innerHTML = "";
+        await renderFlexContextForm(); // im rendring here so te card with the 'border-on-click' class is updated/exists
+        const card = document.getElementById(`${fieldName}-control`);
+        selectFlexField(card);
+    }
+
+    function renderFlexSelectOptions() {
+        flexSelect.innerHTML = `<option value="blank">Select an option</option>`;
+        for (const [key, value] of Object.entries(assetFlexContext)) {
+            if (value === null){
+                flexSelect.innerHTML += `<option value="${key}">${key.split("-").map((word) => word.charAt(0).toUpperCase() + word.slice(1)).join(" ")}</option>`;
+            }
+        }
+        
+    }
+    function renderSelectInfoCards(contextKey, description, isArray=false) {
+        return `
+            <div>
+                <div class="alert alert-warning" role="alert">
```

Comment:
This should not be alert-warning, let's do alert-info. I realize I said light yellow, but that clashed with the other warning, and we should also believe the bootstrap wisdom about what kind of data should be displayed in what way :)